src/win/tty.c

```diff
@@ -1608,6 +1612,30 @@ static int uv_tty_set_cursor_visibility(uv_tty_t* handle,
   return 0;
 }
 
+static int uv_tty_set_cursor_shape(uv_tty_t* handle,
+                                        DWORD* error) {
+  CONSOLE_CURSOR_INFO cursor_info;
+
+  if (!GetConsoleCursorInfo(handle->handle, &cursor_info)) {
+    *error = GetLastError();
+    return -1;
+  }
+
+  if (handle->tty.wr.ansi_csi_argv[0] <= 2) {
+    cursor_info.dwSize = BLOCK_CURSOR;
+  } else {
+    cursor_info.dwSize = UNDERLINE_CURSOR;
+  }
+
+  if (!SetConsoleCursorInfo(handle->handle, &cursor_info)) {
```

Comment:
Please make sure that the cursor shape is properly reset to default in `uv_tty_reset`.